tests/test_ops/test_focal_loss.py

```diff
@@ -57,8 +57,16 @@ def _test_softmax(self, dtype=torch.float):
             assert np.allclose(x.grad.data.cpu(), np_x_grad, 1e-2)
 
     def _test_sigmoid(self, dtype=torch.float):
-        if not torch.cuda.is_available():
-            return
+        if torch.cuda.is_available():
+            device = 'cuda'
+        else:
+            try:
+                if torch.is_mlu_available():
+                    device = 'mlu'
+                else:
+                    return
+            except:
+                return
```

Comment:
We can reuse these lines in other ops.

1. Add `is_cuda` and `is_mlu`

    ```python
    # mmcv/utils/parrots_wrapper.py
    def is_cuda() -> bool:
        return torch.cuda.is_available()
    
    def is_mlu() -> bool:
        if TORCH_VERSION ! = 'parrots':
            try:
                return torch.is_mlu_available()
            except AttributeError:
                return False
        return False
    ```
2. Add above functions to `__init__.py`
https://github.com/open-mmlab/mmcv/blob/c50184630cf300bbf79d7a4b96e30df75eb2e2c3/mmcv/utils/__init__.py#L42-L48

3. Refactor the `_test_sigmoid`

    ```python
    @pytest.mark.parametrize('device_type', [
        pytest.param(
            'cuda',
            marks=pytest.mark.skipif(
                not is_cuda(), reason='requires CUDA support')),
       pytest.param(
            'mlu',
            marks=pytest.mark.skipif(
                not is_mlu(), reason='requires MLU support'))
    ])
    def _test_sigmoid(self, dtype=torch.float, device_type):
    ```